assets/src/edit-story/components/inspector/design/designInspector.js

```diff
@@ -40,9 +42,15 @@ const TopPanels = styled.div`
 const BottomPanels = styled.div``;
 
 function DesignInspector() {
+  const resetStoryAnimationState = useStory(({ actions }) => () =>
+    actions.updateAnimationState({
+      animationState: STORY_ANIMATION_STATE.RESET,
+    })
+  );
+
   return (
     <Wrapper>
-      <TopPanels>
+      <TopPanels onFocus={resetStoryAnimationState}>
```

Comment:
This seemed like the quickest most solid approach to stoping and resetting anything when the user starts interacting with the design panel. Lemme know if you think there's a better approach or this is prone to any edge cases I didn't think of.

Updating the properties on any elements is already blocked at the reducer level when animations are playing or scrubbing, so there's still some safety there even if the user would find a way to interact with this without focusing it first.

Reply:
I'd think it should be "focusin" event instead? That being said, should it pause when we focus into the animations panel itself?

Reply:
Yea I can switch to just adding this as a handler to a `focusin` event in an effect if you'd prefer. Last I checked, react doesn't expose this as a synthetic event, but does differ from the DOM by bubbling up onFocus events which allow the current implementation to work.

Happy to change it regardless, but just for personal learning/growing, what benefits do you see in `focusin` vs `onFocus` for the use case here?

> That being said, should it pause when we focus into the animations panel itself?

This is how I read the current spec. As soon as you start interacting with the design panel we have it reset the animation to the original state/position.

The other option could be resetting the state from the update element properties action in the reducer so it doesn't reset until you actually submit a value. That feels like a more top down approach, which I like, but is a slightly different UX than the current implementation. idk what are your thoughts here?

Reply:
sorry misread that. now noticing you're talking about the animation panel. that's a good point & something I think we should explore and fine tune as we get there. I think this is a good rudimentary approach tho until we get more decisive UX around the animations/animation panel

Reply:
Sounds good. And if `onFocus` indeed bubbles here - great. I thought it doesn't for some reason.